src/Workspaces/Core/Portable/LanguageServices/SyntaxFacts/ISyntaxKindsService_LanguageService.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.LanguageServices
+{
+    /// <summary>
+    /// Provides a uniform view of SyntaxKinds over C# and VB for constructs they have
+    /// in common.
+    /// </summary>
+    internal partial interface ISyntaxKindsService : ILanguageService
```

Comment:
I think we should still consider having this in the shared WorkspaceExtensions layer. CodeFixes themselves can easily get the singleton implementation of this, but it gets pretty messy when we try to move down some extension methods/utilities written in language-agnostic Workspaces layer into the language-agnostic WorkspaceExtensions layer. We need to change all call sites and method chain to thread through the instance of `ISyntaxKindsService` to that helper.

My previous PR (merged yesterday) has already enabled support for `ILanguageService` detection using `document.GetLanguageService` in WorkspaceExtensions.shproj and CodeStyleFixes projects and above. My remaining PR adds MEF based detection of `ILanguageService` using a special CodeStyle layer implementation of HostLanguageServices. I think once that PR goes in, it should be fine to move all the common language services, such as this one, into WorkspaceExtensions.shproj as an `ILanguageService` itself. Does that seem reasonable?

Reply:
> Does that seem reasonable?

So my preference would be to get these things moved down.  Then, when we're ready to have a simple approachthat allows hte services to be used effectively at CodeStyle/CodeStyle.Fixes/Worskspaces layer, we just make that change as appropriately.

I feel like we keep on trying to push too much stuff into really large changes.  I would far prefer smaller, indepedent changes that focus on solving one case.

Reply:
Absolutely, your change as is seems perfectly fine. I was just stating something that can be done in future.

Reply:
Note: i would also strongly prefer not having to have any sort of `#ifdefs` or other mechanisms to work around sharing issues.  (i'm fine with them for versioning issues, like the NRT api stuff).  to me, we should just be tryin to get it so that the files just link in the exact same way into all locations they're needed.

Reply:
Yes, this cleanup is tracked in the follow-up item. Note that those conditional `#ifdefs` will be unavoidable for any code in shared layer that references any Options related type - these are public types on Workspaces and must be linked into the core CodeStyle layer for our IDE code style analyzers to be moved there. This means shared code dealing with Options related types will need to choose between the appropriate definition from these assemblies based on `#if CODE_STYLE` statements. Apart from these files, we should not require any `#ifdefs`

Reply:
SGTM.  Thanks! :)